packages/flutter/lib/src/widgets/platform_menu_bar.dart

```diff
@@ -0,0 +1,886 @@
+    final List<Map<String, Object?>> result = <Map<String, Object?>>[];
+    if (index != 0) {
+      result.add(<String, Object?>{
+        _kIdKey: getId(this),
+        _kIsDividerKey: true,
+      });
+    }
+    for (final MenuItem item in members) {
+      if (item is! DefaultPlatformMenuDelegateSerializable) {
+        throw UnimplementedError('Tried to serialize a menu item that was not a DefaultPlatformMenuDelegateSerializer');
+      }
+      result.addAll((item as DefaultPlatformMenuDelegateSerializable).toChannelRepresentation(
+        delegate,
+        index: index,
+        count: count,
+        getId: getId,
+      ));
+    }
+    if (index != count - 1) {
+      result.add(<String, Object?>{
+        _kIdKey: getId(this),
+        _kIsDividerKey: true,
+      });
+    }
+    return result;
+  }
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties.add(IterableProperty<MenuItem>('members', members));
+  }
+}
+
+/// A class for [MenuItem]s that do not have submenus, but can be selected.
+///
+/// These [MenuItem]s are the leaves of the menu item tree, and [onSelected]
+/// will be called when they are selected by clicking on them, or via an
+/// optional keyboard [shortcut].
+class PlatformMenuItem extends MenuItem with DefaultPlatformMenuDelegateSerializable {
+  /// Creates a const [PlatformMenuItem].
+  ///
+  /// The [label] attribute is required.
+  const PlatformMenuItem({
+    required this.label,
+    this.shortcut,
+    this.onSelected,
+  });
+
+  /// The required label used for rendering the menu item.
+  final String label;
+
+  /// The optional shortcut that selects this [PlatformMenuItem].
+  ///
+  /// This shortcut is only enabled when [onSelected] is set.
+  final ShortcutActivator? shortcut;
+
+  /// An optional callback that is called when this [PlatformMenuItem] is
+  /// selected.
+  ///
+  /// If unset, this menu item will be disabled.
+  final VoidCallback? onSelected;
+
+  @override
+  Iterable<Map<String, Object?>> toChannelRepresentation(
+    DefaultPlatformMenuDelegate delegate, {
+    required int index,
+    required int count,
+    required DefaultPlatformMenuDelegateSerializableIdGenerator getId,
+  }) {
+    return <Map<String, Object?>>[PlatformMenuItem.serialize(this, delegate, getId)];
+  }
+
+  /// Converts the given [PlatformMenuItem] into a data structure accepted by
+  /// the 'flutter/menu' method channel method 'Menu.SetMenu'.
+  ///
+  /// This API is supplied so that implementers of [PlatformMenuItem] can share
+  /// this implementation.
+  static Map<String, Object?> serialize(
+    PlatformMenuItem item,
+    DefaultPlatformMenuDelegate delegate,
+    DefaultPlatformMenuDelegateSerializableIdGenerator getId,
+  ) {
+    int modifiers = 0;
+    int? logicalKeyId;
+    final ShortcutActivator? shortcut = item.shortcut;
+    if (item.shortcut != null && shortcut is SingleActivator) {
```

Comment:
What if shortcut is not a SingleActivator? Should `item.shortcut` just be typed as a SingleActivator?